src/WebView.android.tsx

```diff
@@ -237,9 +237,9 @@ class WebView extends React.Component<AndroidWebViewProps, State> {
     if (onLoadEnd) {
       onLoadEnd(event);
     }
-    if (url === this.startUrl) {
+    if (url && this.startUrl && url.toLowerCase() === this.startUrl.toLowerCase()) {
```

Comment:
I think this could also be written as
```typescript
if (url?.toLowerCase() === this.startUrl?.toLowerCase()) {
```
?

Also I'm curious if comparing URLs via `toLowerCase` is safe for URLs with special characters... (https://stackoverflow.com/a/2140644)

Reply:
Understand that many URLs contain special code, maybe using 
`url.localeCompare(this.startUrl)`
would be a better option

Reply:
Perhaps, although React Native has a rocky history with supporting `locale` functions like that on Android (e.g. https://github.com/facebook/react-native/issues/12597). I'm not sure what the current state of things is, whether it's safe to use in more recent RN versions.